File: mg01/game/tech/Assets/Resources/LuaScript/src/orgu/res/ResMgr.lua
--ResMgr
--@author jr.zeng
--2017年9月29日 下午8:21:09
local modname = "ResMgr"
--==================global reference=======================

--===================namespace========================
local ns = "org"
local using = {--[[namespace1,...]]}
local _ENV = namespace(ns, using)
--===================module property========================
local super = CCModule	--父类
_ENV[modname] = class(modname, super)
local ResMgr = _ENV[modname]

--===================module content========================

function ResMgr:__ctor()

end

function ResMgr:__setup()
    
    AssetCache:setup()
	GameObjCache:setup()
	SprAtlasCache:setup()
	LevelMgr:setup()

end

function ResMgr:__clear()

    AssetCache:clear()
	GameObjCache:clear()
	SprAtlasCache:clear()
	LevelMgr:clear()
    
    
end

function ResMgr:setup_event()



end

function ResMgr:clear_event()


end

--//-------~★~-------~★~-------~★~数据管理~★~-------~★~-------~★~-------//

return ResMgr

File: mg01/game/tech/Assets/Script/Org/Resource/ResMgr.cs
/* ==============================================================================
 * CCResMgrNew
 * @author jr.zeng
 * 2017/5/23 23:17:09
 * ==============================================================================*/

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;


using UnityEngine;
using Object = UnityEngine.Object;

namespace mg.org
{

    public class ResMgr : CCModule
    {
        public ResMgr()
        {
            RegResource();

        }

        //资源注册
        protected virtual void RegResource()
        {
            ResConfig.AddResCfg(CC_RES_ID.RES_REG);
        }


        override protected void __Setup(params object[] params_)
        {
            base.__Setup();

            AssetCache.me.Setup();
            GameObjCache.me.Setup();
            SprAtlasCache.me.Setup();
            SoundCache.me.Setup();
            LevelMgr.me.Setup();
        }

        override protected void __Clear()
        {
            AssetCache.me.Clear();
            GameObjCache.me.Clear();
            SprAtlasCache.me.Clear();
            SoundCache.me.Clear();
            LevelMgr.me.Clear();
        }

        override protected void SetupEvent()
        {

        }

        override protected void ClearEvent()
        {

        }

        //-------∽-★-∽------∽-★-∽--------∽-★-∽数据管理∽-★-∽--------∽-★-∽------∽-★-∽--------//



      
        //-------∽-★-∽------∽-★-∽--------∽-★-∽加载相关∽-★-∽--------∽-★-∽------∽-★-∽--------//

        public Object LoadSync(string resId_, string resName_, object refer)
        {
            string url = ResConfig.GetResUrl(resId_, resName_, ""); //去除后缀名
            Object data = AssetCache.me.LoadSync(url, refer);
            return data;
        }

        public void LoadAsync(string resId_, string resName_, CALLBACK_1 onComplete_, object refer)
        {
            string url = ResConfig.GetResUrl(resId_, resName_, ""); //去除后缀名
            AssetCache.me.LoadAsync(url, onComplete_, refer);
        }

        public void LoadAsync_Prefab(string resId_, string resName_, CALLBACK_GO onComplete_, object refer)
        {
            string url = ResConfig.GetResUrl(resId_, resName_, ""); //去除后缀名
            GameObjCache.me.LoadAsync(url, onComplete_, refer);
        }

        //-------∽-★-∽------∽-★-∽--------∽-★-∽创建相关∽-★-∽--------∽-★-∽------∽-★-∽--------//

        public GameObject LoadGameObj(string resId_, string resName_)
        {
            string url = ResConfig.GetResUrl(resId_, resName_, ""); //去除后缀名
            return GameObjCache.me.LoadSync(url);
        }
        


    }


}
